examples/compare_solvers.jl

```diff
@@ -0,0 +1,38 @@
+## We compare here three different solvers : SCF, direct minimization and newton
+## algorithm.
+
+using DFTK, LinearAlgebra
+
+a = 10.26  # Silicon lattice constant in Bohr
+lattice = a / 2 * [[0 1 1.];
+                   [1 0 1.];
+                   [1 1 0.]]
+Si = ElementPsp(:Si, psp=load_psp("hgh/lda/Si-q4"))
+atoms = [Si => [ones(3)/8, -ones(3)/8]]
+
+model = model_LDA(lattice, atoms)
+kgrid = [3, 3, 3]  # k-point grid (Regular Monkhorst-Pack grid)
+Ecut = 5           # kinetic energy cutoff in Hartree
+basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid)
+tol = 1e-12
+
+## SCF
+scfres_scf = self_consistent_field(basis, tol=tol,
+                                   is_converged=DFTK.ScfConvergenceDensity(tol))
+
+## Direct minimization
+scfres_dm = direct_minimization(basis, tol=tol)
+
+## Newton algorithm
+# start not too far from the solution to ensure convergence : we use here the
+# solution of a single iteration SCF
+scfres_start = self_consistent_field(basis, maxiter=1)
+# remove virtual orbitals
+n_spin = basis.model.n_spin_components
+n_bands = div(div(model.n_electrons, DFTK.filled_occupation(model)), n_spin)
```

Comment:
`div(model.n_electrons, n_spin * DFTK.filled_occupation(model))`

Reply:
Also I guess one could export `filled_occupation`. Sounds like a useful thing to have.

Reply:
We should have a more robust way of dealing with the (possibly unconverged) virtual orbitals returned by the solver at some point anyway

Reply:
There is a function, which can be used to strip them off somewhere if I remember correctly

Reply:
Yes this: `select_eigenpairs_all_kblocks`

Reply:
Yes, but it seems that this function is suited for arguments like `eigres` ? Should we inspire ourselves from this function to create one for a standard psi ?

Reply:
Yeah why not. Give it a shot.